srcpkgs/virtualbox-ose/INSTALL.msg

```diff
@@ -0,0 +1,5 @@
+In kernel 6.12, KVM initializes virtualization on module loading by default.
+This prevents VirtualBox VMs from starting.
+In order to avoid this, either add "kvm.enable_virt_at_load=0" parameter
+into kernel command line or unload corresponding kvm_XXX module.
```

Comment:
Is it really necessary even with vbox 7.1.4? ... I'm not sure how, but in ArchLinux they don't need this kernel parameter --> https://bbs.archlinux.org/viewtopic.php?pid=2210067#p2210067

FYI: voidlinux has been upgraded to kernel 6.12 and virtualbox-ose-7.0.20_1 does not work any longer .. I opened a bug report:

- https://github.com/void-linux/void-packages/issues/53855

Reply:
they add a modprobe.d file for it: https://gitlab.archlinux.org/archlinux/packaging/packages/virtualbox/-/blob/main/virtualbox.modprobe

Reply:
It's definitely necessary. But I'd rather keep it opt-out by default (i.e. manually adding the parameter by the user itself) than making QEMU/KVM buddy annoyed for breaking the exciting new kernel behavior.

Reply:
@Vinfall @classabbyamp thanks a lot for clarification :+1:

Reply:
kernels 6.11 and lower will ignore this option because it was added new. let's also ship a modprobe.d file for this instead of the install message.

Reply:
Adjusted, but I'm not sure if `do_install` is the right place for that. The docs and existing templates have no such case (modprobe) for my reference. Vbox template is also a bit complicated.

Btw I'm still against it as this makes the kernel behave differently than upstream and I am going to be one of the annoyed QEMU/KVM buddy now...

Reply:
it just makes it behave like <=6.11

Reply:
Sure, my point is it should not behave like that, even if it would sacrifice user experience. But it's very personal and I agree adding a modprobe.d conf would benefit others.

Reply:
After rethinking about this, I decided it was acceptable for me. The *upstream* here is actually Oracle VirtualBox, not the kernel. In this regard, this upstream (vbox) explicitly mentioned the solutions in https://github.com/void-linux/void-packages/pull/52910#issuecomment-2481765401, and other distros are already taking the modprobe route. (yeah, I persuaded myself...)

It's still your choice to stick with it but when I do contribution, I also strongly prefer getting things done than anything else. I'll mark the other conversation as resolved.

Reply:
> The _upstream_ here is actually Oracle VirtualBox .. and other distros are already taking the modprobe route

.. I think your arguments are valid :+1:

> (yeah, I persuaded myself...)

as I said, I'm new to void-linux and still in the learning phase :-)

Reply:
> new to void-linux and still in the learning phase

It's fine. Actually I appreciate that someone shares the same attitude because I thought it was a quite marginal issue for others.